src/webrtc/opcodes/SelectProtocol.ts

```diff
@@ -16,50 +16,68 @@
-import SemanticSDP, { MediaInfo, SDPInfo } from "semantic-sdp";
+import MediaServer from "medooze-media-server";
+import SemanticSDP, { MediaInfo } from "semantic-sdp";
+import DefaultSDP from "./sdp.json";
 
 export async function onSelectProtocol(this: WebSocket, payload: Payload) {
-	if (!this.client) return;
+	if (!this.webrtcClient) return;
 
 	const data = validateSchema(
 		"SelectProtocolSchema",
 		payload.d,
 	) as SelectProtocolSchema;
 
-	const offer = SemanticSDP.SDPInfo.parse("m=audio\n" + data.sdp!);
-	this.client.sdp!.setICE(offer.getICE());
-	this.client.sdp!.setDTLS(offer.getDTLS());
+	const offer = SemanticSDP.SDPInfo.parse("m=audio\n" + data.sdp);
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	//@ts-ignore
+	offer.getMedias()[0].type = "audio"; // this is bad, but answer.toString() fails otherwise
+	this.webrtcClient.sdp.setICE(offer.getICE());
+	this.webrtcClient.sdp.setDTLS(offer.getDTLS());
 
-	const transport = endpoint.createTransport(this.client.sdp!);
-	this.client.transport = transport;
-	transport.setRemoteProperties(this.client.sdp!);
-	transport.setLocalProperties(this.client.sdp!);
+	const transport = endpoint.createTransport(this.webrtcClient.sdp);
+	this.webrtcClient.transport = transport;
+	transport.setRemoteProperties(this.webrtcClient.sdp);
+	transport.setLocalProperties(this.webrtcClient.sdp);
 
 	const dtls = transport.getLocalDTLSInfo();
 	const ice = transport.getLocalICEInfo();
-	const port = endpoint.getLocalPort();
+	const port = fs.readFileSync("./tmp/PORT", { encoding: "utf8" }) || 3001;
 	const fingerprint = dtls.getHash() + " " + dtls.getFingerprint();
 	const candidates = transport.getLocalCandidates();
 	const candidate = candidates[0];
 
+	// discord answer
+	/*
+		m=audio 50026 ICE/SDP\n
+		a=fingerprint:sha-256 4A:79:94:16:44:3F:BD:05:41:5A:C7:20:F3:12:54:70:00:73:5D:33:00:2D:2C:80:9B:39:E1:9F:2D:A7:49:87\n
+		c=IN IP4 66.22.206.174\n
+		a=rtcp:50026\n
+		a=ice-ufrag:XxnE\n
+		a=ice-pwd:GLQatPT3Q9dCZVVgVf3J1F\n
+		a=fingerprint:sha-256 4A:79:94:16:44:3F:BD:05:41:5A:C7:20:F3:12:54:70:00:73:5D:33:00:2D:2C:80:9B:39:E1:9F:2D:A7:49:87\n
+		a=candidate:1 1 UDP 4261412862 66.22.206.174 50026 typ host\n
+	*/
+
 	const answer =
-		`m=audio ${port} ICE/SDP` +
-		`a=fingerprint:${fingerprint}` +
-		`c=IN IP4 ${PublicIP}` +
-		`a=rtcp:${port}` +
-		`a=ice-ufrag:${ice.getUfrag()}` +
-		`a=ice-pwd:${ice.getPwd()}` +
-		`a=fingerprint:${fingerprint}` +
-		`a=candidate:1 1 ${candidate.getTransport()} ${candidate.getFoundation()} ${candidate.getAddress()} ${candidate.getPort()} typ host`;
+		`m=audio ${port} ICE/SDP\n` +
+		`a=fingerprint:${fingerprint}\n` +
+		`c=IN IP4 ${PublicIP}\n` +
+		`a=rtcp:${port}\n` +
+		`a=ice-ufrag:${ice.getUfrag()}\n` +
+		`a=ice-pwd:${ice.getPwd()}\n` +
+		`a=fingerprint:${fingerprint}\n` +
+		`a=candidate:1 1 ${candidate.getTransport()} ${candidate.getFoundation()} ${candidate.getAddress()} ${candidate.getPort()} typ host\n`;
```

Comment:
??? why?

Reply:
SDP must be separated by newlines. This is a genuine mistake on my behalf.